bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerBaseHandler.java

```diff
@@ -0,0 +1,169 @@
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    protected State getValue(String channelId, GoEStatusResponseBaseDTO goeResponseBase) {
+        switch (channelId) {
+            case MAX_CURRENT:
+                if (goeResponseBase.maxCurrent == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponseBase.maxCurrent, Units.AMPERE);
+            case CABLE_ENCODING:
+                if (goeResponseBase.cableEncoding == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponseBase.cableEncoding, Units.AMPERE);
+            case FIRMWARE:
+                if (goeResponseBase.firmware == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new StringType(goeResponseBase.firmware);
+            case VOLTAGE_L1:
+                if (goeResponseBase.energy == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponseBase.energy[0], Units.VOLT);
+            case VOLTAGE_L2:
+                if (goeResponseBase.energy == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponseBase.energy[1], Units.VOLT);
+            case VOLTAGE_L3:
+                if (goeResponseBase.energy == null) {
+                    return UnDefType.UNDEF;
+                }
+                return new QuantityType<>(goeResponseBase.energy[2], Units.VOLT);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+        allChannels = getThing().getChannels().stream().map(channel -> channel.getUID().getId())
+                .collect(Collectors.toList());
+
+        logger.debug("Number of channels found: {}", allChannels.size());
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    @Nullable
+    protected GoEStatusResponseBaseDTO getGoEData()
+            throws InterruptedException, TimeoutException, ExecutionException, JsonSyntaxException {
+        return null;
+    }
+
+    protected void updateChannelsAndStatus(@Nullable GoEStatusResponseBaseDTO goeResponse, @Nullable String message) {
+    }
+
+    private void refresh() {
+        synchronized (this) {
+            // Request new GoE data
+            try {
+                GoEStatusResponseBaseDTO goeResponse = getGoEData();
+                updateChannelsAndStatus(goeResponse, null);
+            } catch (InterruptedException | TimeoutException | ExecutionException e) {
+                updateChannelsAndStatus(null, e.getMessage());
```

Comment:
I think this has been commented before. Please handle the excpetions on their own and atleast (for interruptexception) call Thread.currentThread().interrupt();